arch/risc-v/src/common/riscv_swint.c

```diff
@@ -167,6 +175,15 @@ int riscv_swint(int irq, void *context, void *arg)
 {
   uintptr_t *regs = (uintptr_t *)context;
 
+  /* Quick patch to get this same code working with S-mode */
+
+#ifdef CONFIG_ARCH_USE_S_MODE
```

Comment:
I think this is not enough, CURRENT_REGS needs to be cleared later, wihch is done by the interrupt handling logic. Need to re-think.

Reply:
Removed, need to setup CURRENT_REGS in the dispatch logic